test/gtest/test_operations.hpp

```diff
@@ -0,0 +1,110 @@
+                                  dl_module.estMean,
+                                  dl_module.estVariance);
+}
+
+template <typename T>
+void CompareTensor(const tensor<T>& output,
+                   const tensor<T>& ref_out,
+                   const double threshold = std::numeric_limits<T>::epsilon())
+{
+    EXPECT_FALSE(miopen::range_zero(ref_out)) << "CPU data is all zeros";
+    EXPECT_FALSE(miopen::range_zero(output)) << "GPU data is all zeros";
+    EXPECT_FALSE(miopen::find_idx(output, miopen::not_finite) >= 0)
+        << "Non finite number found in the GPU data";
+    EXPECT_TRUE(miopen::range_distance(ref_out) == miopen::range_distance(output));
+    auto error = miopen::rms_range(ref_out, output);
+    EXPECT_FALSE(miopen::find_idx(ref_out, miopen::not_finite) >= 0)
+        << "Non finite number found in the CPU data";
+    EXPECT_TRUE(error < threshold)
+        << "Error beyond tolerance Error:" << error << ",  Threshold: " << threshold;
+}
+} // namespace test
+
+namespace test {
+namespace FusionPlan {
+template <typename DLModule>
+void InitFusionPlan(miopen::FusionPlanDescriptor& fusePlanDesc, DLModule& dl_module)
+{
+    fusePlanDesc = miopen::FusionPlanDescriptor(miopenVerticalFusion, dl_module.GetInputDesc());
+}
+
+template <typename DLModule>
+void AddBnInfer(miopen::FusionPlanDescriptor& fusePlanDesc,
+                miopen::OperatorArgs& params,
+                DLModule& dl_module)
+{
+    auto bnOp = std::make_shared<miopen::BatchNormInferenceFusionOpDescriptor>(dl_module.bn_mode,
+                                                                               dl_module.bn_desc);
+    EXPECT_EQ(fusePlanDesc.AddOp(bnOp), miopenStatusSuccess);
+    bnOp->SetArgs(params,
+                  &dl_module.alpha,
+                  &dl_module.beta,
+                  dl_module.scale_dev.get(),
+                  dl_module.shift_dev.get(),
+                  dl_module.estMean_dev.get(),
+                  dl_module.estVariance_dev.get(),
+                  dl_module.epsilon);
+}
+
+template <typename DLModule>
+void AddActiv(miopen::FusionPlanDescriptor& fusePlanDesc,
+              miopen::OperatorArgs& params,
+              DLModule& dl_module,
+              miopenActivationMode_t activ_mode)
+{
+    auto activOp = std::make_shared<miopen::ActivFwdFusionOpDescriptor>(activ_mode);
+    EXPECT_EQ(fusePlanDesc.AddOp(activOp), miopenStatusSuccess);
+    EXPECT_EQ(activOp->SetArgs(params,
+                               &dl_module.alpha,
+                               &dl_module.beta,
+                               dl_module.activ_alpha,
+                               dl_module.activ_beta,
+                               dl_module.activ_gamma),
+              miopenStatusSuccess);
+}
+
+bool Skip(miopen::Handle& handle)
+{
+    const std::string arch = handle.GetDeviceName();
+    bool skip_test         = (arch != "gfx908" && arch != "gfx90a");
+
+    return skip_test;
```

Comment:
I would suggest we keep these in the tests themselves so that we can document why a test is being skipped inside the test. Centralizing this logic would complicate it very quickly and then make it hard to determine why a test is skipped or failing etc.